DEVELOPMENT.md

```diff
@@ -0,0 +1,188 @@
+  - Tests are implemented in **`./integration_tests/src/tests`** directory.
+  - Some of the test here might be fragile if running along a lot of CPU intensive
+    tasks. That is why **`./.config/nextest.toml`** is limiting concurrency of some tests on CI.
+
+Initially when setting up the repo, or when checking out different branch you need to update
+submodule to the correct version by running
+```
+git submodule update --init --checkout
+```
+
+> Carefully read output of the command to make sure it worked. It will fail if you have uncommitted
+changes in the **`./snapshot_tests/snapshots`** directory.
+
+To run all test run:
+
+```
+cargo nextest run --workspace --profile ci
+```
+
+If you don't have `nextest` installed you can add it with;
+
+```
+cargo install cargo-nextest
+```
+
+To run specific test add name of the function or crate at the end:
+
+e.g.
+
+```
+cargo nextest run --workspace audio_mixing_with_offset
+```
+
+will run a test from **`./integration_tests/src/tests/audio_only.rs`**
+
+#### Updating snapshots
+
+By default, snapshot tests generate new output and compare with the old version committed
+to repo. If version in repo is different or missing then tests will fail.
+
+To generate snapshots for new tests, or update those that should change you need to enable
+`update_snapshots` feature when running tests. It is recommended to only run that command
+for specific tests, especially **`./integration_tests/src/tests`**
+
+e.g.
+
+```
+cargo nextest run --workspace --features update_snapshots audio_mixing_with_offset
+```
+
+will run a test from **./integration_tests/src/tests/audio_only.rs** and if output changed the update the snapshot.
+
+If you made changes that modify the snapshot:
+- Create PR in https://github.com/membraneframework-labs/video_compositor_snapshot_tests repo.
+- Create PR in live compositor repo with link to the snapshot repo PR.
+- Merge them together.
+
+## TypeScript SDK
+
+### Packages
+
+- `live-compositor` - React components and common API types. (analog of `react` package).
+- `@live-compositor/core` - Base implementation that is used by packages like `@live-compositor/node`.
+- `@live-compositor/node` - Node.js SDK for compositor  (analog of `react-dom` for Node.js)
+- `@live-compositor/web-wasm` - Browser SDK for compositor that includes compositor compiled to WASM (analog of `react-dom` for Node.js)
+- `@live-compositor/browser-render` - Rendering ending from LiveCompositor compiled to WASM.
+  - Run `pnpm run build-wasm` to build WASM bundle from Rust code
+  - Run `pnpm run build` to build JS code (WASM has to be build earlier, or use `build:all` in root directory to build everything)
+
+### API compatibility
+
+> WARNING this is the goal, it is currently not true.
+
+SDK should be always compatible with version of code from the repo. If you are changing API the changes to SDK should land in the same PR.
```

Comment:
This is new approach, until now we maintained that SDK should be compatible with released version